test/integration/postgresql-public-iam/controls/pg.rb

```diff
@@ -0,0 +1,136 @@
+    if subject.exit_status == 0
+      JSON.parse(subject.stdout)
+    else
+      {}
+    end
+  end
+
+  describe "postgresql_public_database" do
+    it "global settings are valid" do
+      expect(data['settings']['activationPolicy']).to eq "#{activation_policy}"
+      expect(data['settings']['dataDiskSizeGb']).to eq "#{data_disk_size_gb}"
+      expect(data['settings']['dataDiskType']).to eq "#{data_disk_type}"
+      expect(data['settings']['kind']).to eq "#{kind}"
+      expect(data['settings']['pricingPlan']).to eq "#{pricing_plan}"
+      expect(data['settings']['replicationType']).to eq "#{replication_type}"
+      expect(data['settings']['storageAutoResize']).to eq storage_auto_resize
+      expect(data['settings']['storageAutoResizeLimit']).to eq "#{storage_auto_resize_limit}"
+      expect(data['settings']['tier']).to eq "#{tier}"
+    end
+
+    it "backend type is valid" do
+      expect(data['backendType']).to eq 'SECOND_GEN'
+    end
+
+    it "database versions is valid" do
+      expect(data['databaseVersion']).to eq db_version
+    end
+
+    it "state is valid" do
+      expect(data['state']).to eq 'RUNNABLE'
+    end
+
+    it "region is valid" do
+      expect(data['region']).to eq region
+    end
+
+    it "gce zone is valid" do
+      expect(data['gceZone']).to eq "#{region}-c"
+    end
+
+    it "location preference is valid" do
+      expect(data['settings']['locationPreference']).to include(
+      "kind" => "sql#locationPreference",
+      "zone" => "#{region}-c")
+    end
+
+    it "maintenance window is valid" do
+      expect(data['settings']['maintenanceWindow']).to include(
+      "kind" => "sql#maintenanceWindow",
+      "day" => 1,
+      "hour" => 23,
+      "updateTrack" => "canary")
+    end
+
+    it "database flags are set" do
+      expect(data['settings']['databaseFlags']).to include({
+        "name" => "cloudsql.iam_authentication",
+        "value" => "on"})
+    end
+  end
+
+  describe "Postgres SQL pubic instance" do
+    it "has just one assigned IP address" do
+      expect(data["ipAddresses"].count).to eq(1)
+    end
+
+    it "has expected external IP address" do
+      expect(data["ipAddresses"][0]).to eq(
+        {
+          "type" => "PRIMARY",
+          "ipAddress" => "#{public_ip_address}"
+        }
+      )
+    end
+  end
+end
+
+describe command("gcloud --project='#{project_id}' sql users list --instance #{basename} --format=json") do
+  its(:exit_status) { should eq 0 }
+  its(:stderr) { should eq '' }
+
+  let!(:data) do
+    if subject.exit_status == 0
+      JSON.parse(subject.stdout)
+    else
+      {}
+    end
+  end
+
+  describe "postgresql_public_database" do
+    it "has 1 IAM user" do
+      expect(data.select {|k,v| k['type'] == "CLOUD_IAM_USER"}.size).to eq 1
+    end
+  end
+
+  describe "postgresql_public_database" do
+    it "has 1 IAM service account user" do
+      expect(data.select {|k,v| k['type'] == "CLOUD_IAM_SERVICE_ACCOUNT"}.size).to eq 1
```

Comment:
Let's also assert that our string manipulations within TF are correct, if possible, by expecting a hardcoded expected value?

Reply:
Sure, I can add that check as well.